plugin/declarativenetworkmanager.cpp

```diff
@@ -0,0 +1,281 @@
+    return m_sharedInstance->sessionMode();
+}
+
+uint DeclarativeNetworkManager::inputRequestTimeout() const
+{
+    return m_sharedInstance->inputRequestTimeout();
+}
+
+bool DeclarativeNetworkManager::servicesEnabled() const
+{
+    return m_sharedInstance->servicesEnabled();
+}
+
+void DeclarativeNetworkManager::setServicesEnabled(bool enabled)
+{
+    m_sharedInstance->setServicesEnabled(enabled);
+}
+
+bool DeclarativeNetworkManager::technologiesEnabled() const
+{
+    return m_sharedInstance->technologiesEnabled();
+}
+
+void DeclarativeNetworkManager::setTechnologiesEnabled(bool enabled)
+{
+    m_sharedInstance->setTechnologiesEnabled(enabled);
+}
+
+bool DeclarativeNetworkManager::isValid() const
+{
+    return m_sharedInstance->isValid();
+}
+
+bool DeclarativeNetworkManager::connected() const
+{
+    return m_sharedInstance->connected();
+}
+
+bool DeclarativeNetworkManager::connecting() const
+{
+    return m_sharedInstance->connecting();
+}
+
+QString DeclarativeNetworkManager::wifiTechnologyPath() const
+{
+    return m_sharedInstance->wifiTechnologyPath();
+}
+
+QString DeclarativeNetworkManager::cellularTechnologyPath() const
+{
+    return m_sharedInstance->cellularTechnologyPath();
+}
+
+QString DeclarativeNetworkManager::bluetoothTechnologyPath() const
+{
+    return m_sharedInstance->bluetoothTechnologyPath();
+}
+
+QString DeclarativeNetworkManager::gpsTechnologyPath() const
+{
+    return m_sharedInstance->gpsTechnologyPath();
+}
+
+QString DeclarativeNetworkManager::ethernetTechnologyPath() const
+{
+    return m_sharedInstance->ethernetTechnologyPath();
+}
+
+QStringList DeclarativeNetworkManager::servicesList(const QString &tech)
+{
+    return m_sharedInstance->servicesList(tech);
+}
+
+QStringList DeclarativeNetworkManager::savedServicesList(const QString &tech)
+{
+    return m_sharedInstance->savedServicesList(tech);
+}
+
+QStringList DeclarativeNetworkManager::availableServices(const QString &tech)
+{
+    return m_sharedInstance->availableServices(tech);
+}
+
+QStringList DeclarativeNetworkManager::technologiesList()
+{
+    return m_sharedInstance->technologiesList();
+}
+
+QString DeclarativeNetworkManager::technologyPathForService(const QString &path)
+{
+    return m_sharedInstance->technologyPathForService(path);
+}
+
+QString DeclarativeNetworkManager::technologyPathForType(const QString &type)
+{
+    return m_sharedInstance->technologyPathForType(type);
+}
+
+void DeclarativeNetworkManager::resetCountersForType(const QString &type)
+{
+    m_sharedInstance->resetCountersForType(type);
+}
+
+void DeclarativeNetworkManager::setOfflineModeProperty(bool offlineMode)
+{
+    m_sharedInstance->setOfflineMode(offlineMode);
+}
+
+void DeclarativeNetworkManager::setSessionModeProperty(bool sessionMode)
+{
+    m_sharedInstance->setSessionMode(sessionMode);
+}
+
+void DeclarativeNetworkManager::setOfflineMode(bool offlineMode)
+{
+    qWarning() << "NetworkManager:setOfflineMode() is deprecated. Assign to property";
```

Comment:
Would it be good if we'd also report these at build time to properly deprecate these in the header? Like
```
QT_DEPRECATED_X("use setOfflineModeProperty instead") void setOfflineMode();
```

Reply:
That doesn't work, it requires depending code to opt-in for Qt deprecation warnings.

Q_DECL_DEPRECATED_X() works nicer, but then it warns also on the internal use on the factory stuff, also from moc_ .cpp. With some #ifdeffing I could avoid it on internal builds but not sure about that moc_. Dunno, I can do some more investigation and perhaps mark some parts as deprecated, but I'd like to also keep this simple and the build of connman-qt without warnings.

Reply:
The simper, the better. I was just thinking of that with build warnings we'd eventually get rid of the deprecated stuff. I don't mind if we'd do a follow-up task on that to properly do the deprecation with builds. It is fine as it is now.

Reply:
I added the Q_DECL_DEPRECATED for what I could without getting warnings on building the project itself, mainly not included now in things that have Q_PROPERTY. Speaking of which, now that the qml api is separated, having them on the c++ class doesn't have that much purpose anymore. Not that it harms much either.

While at it, I also made the session mode thing deprecated no-op.

This particular place not touched with deprecations since it's the qml api. Can have only runtime warning. Though could later bump the module import version number and avoid deprecated api with that.

Reply:
Looks ok to me. And there was this https://github.com/sailfishos/nemo-qml-plugin-connectivity/pull/10#issuecomment-2025171932 pointing out the effect.